syntax/resolve.rs

```diff
@@ -1,10 +1,12 @@
-use crate::syntax::{Lifetimes, NamedType, Pair, Types};
+use crate::syntax::{Derive, Lifetimes, NamedType, Pair, Types};
 use proc_macro2::Ident;
 
 #[derive(Copy, Clone)]
 pub struct Resolution<'a> {
```

Comment:
This change I was also uncertain of. I'm unsure the intended scope of `Resolution`, but it seemed to me that bounds and derives were as associated to a type as `generics`. Happy to split this out to a separate collection in `Types`